gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/TaskManager.java

```diff
@@ -243,8 +246,18 @@ private Set<Task> removeProcessFileFromTasks(ProcessFile processFile) {
             .collect(Collectors.toSet());
     }
 
+    /**
+     * We compare the size of inputs list from process files of the task and the size of inputs from configuration.
+     * If its equal task is ready otherwise it is created. When it is null it is not created.
+     * This works because we consider there are only one file type per inputs. We call this method at adding and
+     * deletion to check if the status has changed.
+     *
+     * @param task: Task on which to evaluate the status.
+     */
     private void checkAndUpdateTaskStatus(Task task) {
-        int fileNumber = task.getProcessFiles().size();
+        long fileNumber = task.getProcessFiles().stream()
+                .filter(processFile -> MinioAdapterConstants.DEFAULT_GRIDCAPA_INPUT_GROUP_METADATA_VALUE.equals(processFile.getFileGroup()))
+                .count();
```

Comment:
As we could add a input file that would not be part of the input list, I think the check should be more accurate here. Check that the list of inputs in config in fulfilled.